updating with new JEP process

choldgraf opened the pull request:
This is one step towards implementing the meta-JEP from https://github.com/jupyter/enhancement-proposals/pull/29

It does the following:

* Updates the [enhancement proposal guidelines](https://github.com/jupyter/enhancement-proposals/blob/master/jupyter-enhancement-proposal-guidelines/jupyter-enhancement-proposal-guidelines.md) with text that was taken from the PR
  - I made minor modifications for structure and cohesion, but none that should have changed the intent or meaning of the meta-JEP.
* Adds a new JEP proposal template. This is heavily-inspired by the [Rust RFC template](https://github.com/rust-lang/rfcs/blob/master/0000-template.md)
* Adds a PR template to remind people to follow the JEP process.

Happy to take comments or suggestions!

captainsafia commented on jupyter-enhancement-proposal-guidelines/jupyter-enhancement-proposal-guidelines.md:
```diff
@@ -1,61 +1,201 @@
+                                                    |
++--------------+   +-----------+              +-----+-----+             +-------------+   +-----------+
+|              |   |           |              |           |             |             |   |           |
+| pre-proposal +---> submitted +-------------->  RFC/FCP  +-------------> in progress +---> completed |
```
Nit: Perhaos make this `RFC (Request for Comments)` for people who might be unfamiliar with the term.

choldgraf replied:
+1 from me, OK if I leave out the "FCP" in that case, to save space?

captainsafia replied:
Yep -- sounds reasonable.

captainsafia commented on jupyter-enhancement-proposal-guidelines/jupyter-enhancement-proposal-guidelines.md:
````diff
@@ -1,61 +1,201 @@
+```
+                                                                    +-----------+
+                                                                    |           |
+                                              +-----------+         | withdrawn |
````
Do we have an arrow that takes you from one (or more) of the steps to withdrawn?

choldgraf replied:
yeah I wondered about this too - I wasn't sure how to do this without having a bunch of arrows cluttering up the viz. Maybe we can have a "floating arrow" that doesn't come directly from one box to make it clear that at any point it can move into "withdrawn"?

captainsafia replied:
Floating arrow works. Or and endnote in the document that lets people know you can move from any state to withdrawn. Endnote seems like it would be the least obtrusive and most informative.

choldgraf commented:
OK, just added a commit that addresses both of your comments. For the "withdrawn" section, I added a short note in the diagram, as well as note at the beginning of the "phases" section of the document

captainsafia commented:
Sounds good -- thanks.

choldgraf commented:
Hey all - comments addressed. Ready to merge from my perspective (note this should be seen as one PR implementing JEP #29 , and more can follow if there are other things to improve), or happy to address more comments!

choldgraf commented:
wohoo! thanks all for your comments

damianavila commented on jupyter-enhancement-proposal-guidelines/jupyter-enhancement-proposal-guidelines.md:
```diff
@@ -1,61 +1,205 @@
+At the end of the FCP, the JEP is either:
+
+* **approved**, in which case the PR is merged and work may commence on implementing the JEP (see Phase 3, below)
+* **rejected**, in which case the PR is closed
```
Thanks!
